components/header/timeframe_selector_mobile.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState, useContext } from 'react';
+import styled, { css } from 'styled-components';
+import ReactModal from 'react-modal';
+
+import { TimeframeNames } from '@components/header';
+import { BoxWrapper } from '@components/box';
+import { ArrowDown, Checkmark } from '@components/icons';
+import { CloseModalButton } from '@components/header/wallet_modal.style';
+
+import type { TimeframeSelectorOptions } from '@contexts/timeframe_selector';
+import TimeframeContext from '@contexts/timeframe_selector';
+
+import { device } from '@utils';
+
+const TimeframeSelectorMobileWrapper = styled(BoxWrapper)`
+	flex-direction: row;
+	justify-content: space-between;
+	padding-left: 1rem;
+	padding-right: 1rem;
+
+	@media ${device.tablet} {
+		display: none;
+	}
+`;
+
+const TimeframeButtonWrapper = styled.div`
+	display: grid;
+	height: 100%;
+`;
+
+type TimeframeButtonProps = {
+	selected?: boolean;
+};
+const TimeframeButton = styled.button<TimeframeButtonProps>`
+	all: unset;
+	font-size: 1.125rem;
+	font-weight: 400;
+	cursor: pointer;
+	display: flex;
+	align-items: center;
+	justify-content: space-between;
+	background: ${({ selected }) => (selected ? '#1D1D1D' : 'transparent')};
+	padding: 2rem 2.5rem;
+
+	${({ selected }) =>
+		!selected &&
+		css`
+			& > svg {
+				display: none;
+			}
+		`}
+`;
+
+const TimeframeText = styled.p`
+	font-size: 0.75rem;
+	& > span {
+		font-weight: 700;
+	}
+`;
+
+const TimeframeSelectorMobileModal = ({ isOpen, closeModal }: { isOpen: boolean; closeModal: Function }) => {
+	const [timeframe, setTimeframe] = useContext(TimeframeContext);
+	const selectTimeframe = (timeframe: TimeframeSelectorOptions) => {
+		setTimeframe(timeframe);
+		closeModal();
+	};
+
+	return (
+		<ReactModal
+			isOpen={isOpen}
+			contentLabel='Select timeframe'
+			onRequestClose={() => closeModal()}
+			className='react-modal-content'
+			overlayClassName='react-modal-overlay'
+		>
+			<CloseModalButton onClick={() => closeModal()}>Close Modal</CloseModalButton>
+			<TimeframeButtonWrapper>
+				<TimeframeButton selected={timeframe === 'weekly'} onClick={() => selectTimeframe('weekly')}>
+					{TimeframeNames['weekly']}
+					<Checkmark />
+				</TimeframeButton>
+				<TimeframeButton selected={timeframe === 'lastWeek'} onClick={() => selectTimeframe('lastWeek')}>
+					{TimeframeNames['lastWeek']}
+					<Checkmark />
+				</TimeframeButton>
+				<TimeframeButton selected={timeframe === 'total'} onClick={() => selectTimeframe('total')}>
+					{TimeframeNames['total']}
+					<Checkmark />
+				</TimeframeButton>
+			</TimeframeButtonWrapper>
```

Comment:
Seems to me that there could be a `Timeframe` component with a `isMobile` prop to conditionally render the checkmarks. Unless I'm missing some other differences here

Reply:
You're right :)

Reply:
The CSS differences are notable though. It would require more media checks to merge it all